### ISSUE TITLE
mage: maybe break into two packages

### ISSUE THREAD

**Issue by sckott** [jonesor/compadreDB#6]
on call with @robcito  and @tdjames1  recently, we talked about possibly splitting `Mage` into two packages, 

* one focusing on interacting with compadre/comadre and other databases
* one focusing on general purpose set of functions for working with the data above

thoughts, feeling, etc.

**Comment by simonrolph** [comment]
Hi all,

I agree with the split. Blog post coming soon that will explain more but the 'next-gen' COMPADRE will have API (bare bones guide on how it will work here: https://github.com/Spandex-at-Exeter/demography_database/wiki/API-guide). So there will be R functions to interact with that. We will also offer a 'download a csv' functionality but we may want R functions to transform to Rdata object / query the csv etc as with the Rdata object.

**Comment by iainmstott** [comment]
I agree with the split too (and have discussed with @jonesor before). A package focussing on general functions could also be useful for matrices which aren't compadre. Owen and I (and maybe @robcito?) had a conversation where we talked about a more life-history-aligned package (Mage covers a lot of this already, e.g. makeLifeTable, kentropy, lifeTimeRepEvents functions), whilst a lot of the population dynamics stuff (with more eventually to come) covered by popdemo (and popdemo will be tweaked to work better with compadre).

I think Simon has it exactly right, a compadre package would be best if it was limited to transforming/subsetting/querying the data.
